source/base/tensor_function_parser.cc

```diff
@@ -0,0 +1,419 @@
+      tfp.get()[component]->DefineFun("log", internal::mu_log, true);
+      tfp.get()[component]->DefineFun("pow", internal::mu_pow, true);
+      tfp.get()[component]->DefineFun("erfc", internal::mu_erfc, true);
+      tfp.get()[component]->DefineFun("rand_seed", internal::mu_rand_seed, true);
+      tfp.get()[component]->DefineFun("rand", internal::mu_rand, true);
+
+      try
+        {
+          // muparser expects that functions have no
+          // space between the name of the function and the opening
+          // parenthesis. this is awkward because it is not backward
+          // compatible to the library we used to use before muparser
+          // (the tfparser library) but also makes no real sense.
+          // consequently, in the expressions we set, remove any space
+          // we may find after function names
+          std::string transformed_expression = expressions[component];
+
+          const char *function_names[] = {// functions predefined by muparser
+                                          "sin",
+                                          "cos",
+                                          "tan",
+                                          "asin",
+                                          "acos",
+                                          "atan",
+                                          "sinh",
+                                          "cosh",
+                                          "tanh",
+                                          "asinh",
+                                          "acosh",
+                                          "atanh",
+                                          "atan2",
+                                          "log2",
+                                          "log10",
+                                          "log",
+                                          "ln",
+                                          "exp",
+                                          "sqrt",
+                                          "sign",
+                                          "rint",
+                                          "abs",
+                                          "min",
+                                          "max",
+                                          "sum",
+                                          "avg",
+                                          // functions we define ourselves above
+                                          "if",
+                                          "int",
+                                          "ceil",
+                                          "cot",
+                                          "csc",
+                                          "floor",
+                                          "sec",
+                                          "pow",
+                                          "erfc",
+                                          "rand",
+                                          "rand_seed"};
```

Comment:
Could we move this array to the muparser header?

Reply:
I moved it to the header. I also changed that in function_parser.cc